app/src/main/java/com/example/tasky/authentication/domain/EmailValidator.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.tasky.authentication.domain
+
+import android.util.Patterns
+import javax.inject.Inject
+
+interface EmailValidator {
```

Comment:
This interface its in domain. Does the implementation (since is "business logic") belongs to domain right?

Reply:
Implementation should be on Data
Since it uses Android stuff

Same with the Implementation of Repository for example

Interface on domain
Implementation on data (since it accesses Room and Retrofit)